### PR TITLE
tests: add a server test for AWS EC2

### LINKED ISSUE #17: tests: AWS integration test harness
Much like tests/gcp, add an integration test that runs on EC2 and checks that traces and logs are written.

### PR DESCRIPTION
**Pull request by shantuo**
Fixes #17

### CODE REVIEW

File: tests/aws/app/app_test.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	}
+	session, err := c.NewSession()
+	if err != nil {
+		return fmt.Errorf("failed to create session: %v", err)
+	}
+
+	// This close the session channel and the TCP connection, but doesn't terminate
+	// the running app. session.Signal() does not terminate it either, see
+	// https://github.com/golang/go/issues/4115. This is OK for our use case since
+	// we tear it down after the test finishes.
```
Comment (zombiezen) [inline tests/aws/app/app_test.go]: tear down the instance

File: tests/aws/app/app_test.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+	defer func() {
+		session.Close()
+		c.Close()
+	}()
+	return session.Start("/home/admin/app 2>&1 | logger")
+}
+
+func publicKeyFile(path string) (ssh.AuthMethod, error) {
```
Comment (zombiezen) [inline tests/aws/app/app_test.go]: This function needs a more clear name, as the path you are giving is for a private key file. 🙂

File: tests/aws/app/app_test.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+		t.Fatalf("error creating logadmin client: %v", err)
+	}
+	defer c.Close()
+
+	if err := testutil.Retry(t, readLogEntries(ctx, c, p)); err != nil {
+		t.Error(err)
+	}
+}
+
+func readLogEntries(ctx context.Context, c *logadmin.Client, u string) func() error {
```
Comment (zombiezen) [inline tests/aws/app/app_test.go]: If you want to split this out into a separate function, don't return a `func() error`. Perform the code directly in the function, and change the caller to do the adapting into the right kind of closure.

File: tests/aws/app/app_test.go [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+		log.Fatalf("error creating an AWS session: %v\n", err)
+	}
+	c := xray.New(sess)
+	if err := testutil.Retry(t, readTrace(c, p, startTime)); err != nil {
+		t.Error(err)
+	}
+}
+
+func readTrace(c *xray.XRay, u string, startTime time.Time) func() error {
+	return func() error {
```
Comment (zombiezen) [inline tests/aws/app/app_test.go]: (Same note here regarding closures.)

File: tests/aws/app/app_test.go (hunk 1 above)
Comment (shantuo) [reply tests/aws/app/app_test.go]: Done.

File: tests/aws/app/app_test.go (hunk 2 above)
Comment (shantuo) [reply tests/aws/app/app_test.go]: Split the PublicKeys part out, renamed to signerFromFile.

File: tests/aws/app/app_test.go (hunk 3 above)
Comment (shantuo) [reply tests/aws/app/app_test.go]: Done.

File: tests/aws/app/app_test.go (hunk 4 above)
Comment (shantuo) [reply tests/aws/app/app_test.go]: Done.

**Pull request merged**